src/main/java/bio/terra/service/dataset/DatasetBucketDao.java

```diff
@@ -85,7 +85,8 @@ public void decrementDatasetBucketLink(UUID datasetId, UUID bucketResourceId) {
         datasetBucketLinkUpdate(sqlDecrementCount, datasetId, bucketResourceId);
     }
 
-    private boolean datasetBucketLinkExists(UUID datasetId, UUID bucketResourceId) {
+    // public for testing purposes
```

Comment:
There's a `@VisisbleForTesting` annotation that we should use for that.  Also, I think you can get away with making it package private (e.g. just take out the `public`)

Reply:
Oh, ok! I removed "public'.